planex/cmd/clone.py

```diff
@@ -216,9 +219,10 @@ def extract(url_str, destination):
         fetch_url(url, archive_path, 5)
 
     # extract the archive
-    tara = tarfile.open(archive_path)
-    tara.extractall(str(destination))
-    tara.close()
+    if tarfile.is_tarfile(archive_path):
+        tara = tarfile.open(archive_path)
```

Comment:
```
with tarfile.open(archive_path) as tara:
    tara.extractall(str(destination)
```
??

Reply:
oh I was completely on autopilot and didn't even take that in. Yeah there's a tons of open and closes from someone unfamiliar with context handlers.